File: Problem-039/solution.py
def int_right_triangles(n):
    max_solutions = 0
    result = 0

    # Check every possible perimeter
    for p in range(1, n + 1):
        count = 0

        # Try possible sides a and b
        for a in range(1, p // 3):
            for b in range(a + 1, p // 2):
                c = p - a - b

                # Check Pythagorean theorem
                if a * a + b * b == c * c:
                    count += 1

        # Keep the perimeter with the most solutions
        if count > max_solutions:
            max_solutions = count
            result = p

    return result


print(int_right_triangles(1000))

File: Problem-039/solution.js
function isRightTriangle(n){
  let maxSolution = 0 ;
  let result = 0;
  for (let p = 1;p<=n;p++){
    let count = 0;
    for(let a = 1; a<p/3; a++){
      for (let b = a+1; b<p/2; b++){
        let c =p-a-b;
        if (a*a+b*b === c*c){
          count ++;
        }
      }
    }
    if ( conut > maxSolution){
      maxSolution = count;
      result = p;
    }
  }
  return result;
}
